docs/en/sql-reference/functions/other-functions.md

```diff
@@ -998,17 +998,110 @@ SELECT version()
 Returns the build ID generated by a compiler for the running ClickHouse server binary.
 If executed in the context of a distributed table, this function generates a normal column with values relevant to each shard. Otherwise it produces a constant value.
 
-## blockNumber()
+## blockNumber
 
-Returns the sequence number of the data block where the row is located.
+Returns the sequence number of the [data block](https://clickhouse.com/docs/en/integrations/python#data-blocks) where the row is located.
 
-## rowNumberInBlock() {#rowNumberInBlock}
+**Syntax**
+
+```sql
+blockNumber()
+```
+
+**Returned value**
+
+- Sequence number of the data block where the row is located. [UInt64](../data-types/int-uint.md).
+
+**Example**
+
+Query:
+
+```sql
+SELECT groupUniqArray(blockNumber())
+FROM
+(
+    SELECT *
+    FROM system.numbers_mt
+    LIMIT 100000
+)
+```
+
+Result:
+
+```response
+┌─groupUniqArray(blockNumber())─┐
+│ [6,7]                         │
+└───────────────────────────────┘
+```
+
+## rowNumberInBlock {#rowNumberInBlock}
+
+Returns the ordinal number of the row in the [data block](https://clickhouse.com/docs/en/integrations/python#data-blocks). Different data blocks are always recalculated.
+
+**Syntax**
+
+```sql
+rowNumberInBlock()
+```
+
+**Returned value**
+
+- Ordinal number of the row in the data block starting from 0. [UInt64](../data-types/int-uint.md).
+
+**Example**
+
+Query:
+
+```sql
+SELECT
+    min(rowNumberInBlock()),
+    max(rowNumberInBlock())
+FROM (SELECT * FROM system.numbers_mt LIMIT 100000);
+```
+
+Result:
+
+```response
```

Comment:
Not a useful example, let's do

```sql
Not a useful example, let's do

```sql
SELECT rowNumberInBlock()
FROM
(
    SELECT *
    FROM system.numbers_mt
    LIMIT 10
)
SETTINGS max_block_size = 2
```